{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyN0vRmM6VdGOpAWHctmSWoO",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/AhmedZaghl0ul/Machine-Learning/blob/main/Prostate_Cancer.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "15c9da53",
        "outputId": "a60f7b07-0d20-4de3-b754-1af1b80afc6e"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "cd /content/drive/MyDrive/DataSets"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GiNE73f1TYm2",
        "outputId": "53a99467-f67e-4a20-cdd7-6827c1db38ce"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive/DataSets\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "dir = 'prad_mcspc_mskcc_2020_clinical_data.tsv'"
      ],
      "metadata": {
        "id": "HO69I-G7Tn6y"
      },
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd"
      ],
      "metadata": {
        "id": "rjYSP_UUT0Z8"
      },
      "execution_count": 5,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df = pd.read_csv(dir, sep='\\t')"
      ],
      "metadata": {
        "id": "Nj1p5HU7TwTH"
      },
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "classes_name = ['high volume' , 'low volume']"
      ],
      "metadata": {
        "id": "Xbd10oILT2xB"
      },
      "execution_count": 7,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 499
        },
        "id": "fjQ84IPyUiNk",
        "outputId": "3f759bb7-535b-4fea-c7e1-a2cb41b75059"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                Study ID Patient ID          Sample ID  \\\n",
              "0  prad_mcspc_mskcc_2020  P-0000373  P-0000373-T01-IM3   \n",
              "1  prad_mcspc_mskcc_2020  P-0000431  P-0000431-T01-IM3   \n",
              "2  prad_mcspc_mskcc_2020  P-0000442  P-0000442-T01-IM3   \n",
              "3  prad_mcspc_mskcc_2020  P-0000509  P-0000509-T01-IM3   \n",
              "4  prad_mcspc_mskcc_2020  P-0000541  P-0000541-T01-IM3   \n",
              "\n",
              "  Age at Sample Collection Androgen Deprivation Therapy (ADT) Archer Panel  \\\n",
              "0                       55                                 No           NO   \n",
              "1                       57                                Yes           NO   \n",
              "2                       67                                 No           NO   \n",
              "3                       64                                Yes           NO   \n",
              "4                       57                                Yes           NO   \n",
              "\n",
              "  Biopsy Gleason Grade      Cancer Type     Cancer Type Detailed  \\\n",
              "0                 9-10  Prostate Cancer  Prostate Adenocarcinoma   \n",
              "1                    8  Prostate Cancer  Prostate Adenocarcinoma   \n",
              "2                    8  Prostate Cancer  Prostate Adenocarcinoma   \n",
              "3                 9-10  Prostate Cancer  Prostate Adenocarcinoma   \n",
              "4                 9-10  Prostate Cancer  Prostate Adenocarcinoma   \n",
              "\n",
              "  Timing of metastases  ... Number of Samples Per Patient Sample coverage  \\\n",
              "0   De-novo metastatic  ...                             1             708   \n",
              "1   De-novo metastatic  ...                             1             530   \n",
              "2   De-novo metastatic  ...                             1             922   \n",
              "3   De-novo metastatic  ...                             1             651   \n",
              "4   De-novo metastatic  ...                             1             515   \n",
              "\n",
              "    Sex Time from sample to castration resistance [months] Somatic Status  \\\n",
              "0  Male                                              20.03        Matched   \n",
              "1  Male                                              35.58        Matched   \n",
              "2  Male                                              56.29        Matched   \n",
              "3  Male                                              39.84        Matched   \n",
              "4  Male                                              21.58        Matched   \n",
              "\n",
              "  SO comments Survival Status  Tissue Site TMB (nonsynonymous)  Tumor Purity  \n",
              "0         NaN            Dead   Lymph Node            1.109155          60.0  \n",
              "1         NaN           Alive   Lymph Node            1.109155          60.0  \n",
              "2         NaN           Alive     Prostate            1.109155          50.0  \n",
              "3         NaN           Alive     Prostate            2.218311          50.0  \n",
              "4         NaN            Dead     Prostate            3.327466          60.0  \n",
              "\n",
              "[5 rows x 40 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-6bf2ed09-2d32-414c-b5b5-b1527f8c96f0\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Study ID</th>\n",
              "      <th>Patient ID</th>\n",
              "      <th>Sample ID</th>\n",
              "      <th>Age at Sample Collection</th>\n",
              "      <th>Androgen Deprivation Therapy (ADT)</th>\n",
              "      <th>Archer Panel</th>\n",
              "      <th>Biopsy Gleason Grade</th>\n",
              "      <th>Cancer Type</th>\n",
              "      <th>Cancer Type Detailed</th>\n",
              "      <th>Timing of metastases</th>\n",
              "      <th>...</th>\n",
              "      <th>Number of Samples Per Patient</th>\n",
              "      <th>Sample coverage</th>\n",
              "      <th>Sex</th>\n",
              "      <th>Time from sample to castration resistance [months]</th>\n",
              "      <th>Somatic Status</th>\n",
              "      <th>SO comments</th>\n",
              "      <th>Survival Status</th>\n",
              "      <th>Tissue Site</th>\n",
              "      <th>TMB (nonsynonymous)</th>\n",
              "      <th>Tumor Purity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>prad_mcspc_mskcc_2020</td>\n",
              "      <td>P-0000373</td>\n",
              "      <td>P-0000373-T01-IM3</td>\n",
              "      <td>55</td>\n",
              "      <td>No</td>\n",
              "      <td>NO</td>\n",
              "      <td>9-10</td>\n",
              "      <td>Prostate Cancer</td>\n",
              "      <td>Prostate Adenocarcinoma</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>...</td>\n",
              "      <td>1</td>\n",
              "      <td>708</td>\n",
              "      <td>Male</td>\n",
              "      <td>20.03</td>\n",
              "      <td>Matched</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Dead</td>\n",
              "      <td>Lymph Node</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>60.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>prad_mcspc_mskcc_2020</td>\n",
              "      <td>P-0000431</td>\n",
              "      <td>P-0000431-T01-IM3</td>\n",
              "      <td>57</td>\n",
              "      <td>Yes</td>\n",
              "      <td>NO</td>\n",
              "      <td>8</td>\n",
              "      <td>Prostate Cancer</td>\n",
              "      <td>Prostate Adenocarcinoma</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>...</td>\n",
              "      <td>1</td>\n",
              "      <td>530</td>\n",
              "      <td>Male</td>\n",
              "      <td>35.58</td>\n",
              "      <td>Matched</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Alive</td>\n",
              "      <td>Lymph Node</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>60.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>prad_mcspc_mskcc_2020</td>\n",
              "      <td>P-0000442</td>\n",
              "      <td>P-0000442-T01-IM3</td>\n",
              "      <td>67</td>\n",
              "      <td>No</td>\n",
              "      <td>NO</td>\n",
              "      <td>8</td>\n",
              "      <td>Prostate Cancer</td>\n",
              "      <td>Prostate Adenocarcinoma</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>...</td>\n",
              "      <td>1</td>\n",
              "      <td>922</td>\n",
              "      <td>Male</td>\n",
              "      <td>56.29</td>\n",
              "      <td>Matched</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Alive</td>\n",
              "      <td>Prostate</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>50.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>prad_mcspc_mskcc_2020</td>\n",
              "      <td>P-0000509</td>\n",
              "      <td>P-0000509-T01-IM3</td>\n",
              "      <td>64</td>\n",
              "      <td>Yes</td>\n",
              "      <td>NO</td>\n",
              "      <td>9-10</td>\n",
              "      <td>Prostate Cancer</td>\n",
              "      <td>Prostate Adenocarcinoma</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>...</td>\n",
              "      <td>1</td>\n",
              "      <td>651</td>\n",
              "      <td>Male</td>\n",
              "      <td>39.84</td>\n",
              "      <td>Matched</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Alive</td>\n",
              "      <td>Prostate</td>\n",
              "      <td>2.218311</td>\n",
              "      <td>50.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>prad_mcspc_mskcc_2020</td>\n",
              "      <td>P-0000541</td>\n",
              "      <td>P-0000541-T01-IM3</td>\n",
              "      <td>57</td>\n",
              "      <td>Yes</td>\n",
              "      <td>NO</td>\n",
              "      <td>9-10</td>\n",
              "      <td>Prostate Cancer</td>\n",
              "      <td>Prostate Adenocarcinoma</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>...</td>\n",
              "      <td>1</td>\n",
              "      <td>515</td>\n",
              "      <td>Male</td>\n",
              "      <td>21.58</td>\n",
              "      <td>Matched</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Dead</td>\n",
              "      <td>Prostate</td>\n",
              "      <td>3.327466</td>\n",
              "      <td>60.0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 40 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-6bf2ed09-2d32-414c-b5b5-b1527f8c96f0')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-6bf2ed09-2d32-414c-b5b5-b1527f8c96f0 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-6bf2ed09-2d32-414c-b5b5-b1527f8c96f0');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-ed311600-0bff-4133-8d1e-524f4c24bf00\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ed311600-0bff-4133-8d1e-524f4c24bf00')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-ed311600-0bff-4133-8d1e-524f4c24bf00 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df"
            }
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df.columns"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GoNUGRVPUk5Q",
        "outputId": "4da47367-50fc-4e11-921d-82dfaef04b80"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Index(['Study ID', 'Patient ID', 'Sample ID', 'Age at Sample Collection',\n",
              "       'Androgen Deprivation Therapy (ADT)', 'Archer Panel',\n",
              "       'Biopsy Gleason Grade', 'Cancer Type', 'Cancer Type Detailed',\n",
              "       'Timing of metastases', 'Disease Extent', 'Disease volume',\n",
              "       'Fraction Genome Altered', 'Gene Panel', 'Institute Source',\n",
              "       'Castration Resistance Event', 'MSI Comment', 'MSI Score', 'MSI Type',\n",
              "       'MSK Slide ID', 'Mutation Count', 'Oncotree Code',\n",
              "       'Overall Survival since Sample Collection (Months)',\n",
              "       '12-245 Part A Consented', '12-245 Part C Consented',\n",
              "       'MSK Pathology Slide Available', 'Pediatric Case Indicator',\n",
              "       'Prostate-specific antigen', 'Race Category', 'Sample Class',\n",
              "       'Number of Samples Per Patient', 'Sample coverage', 'Sex',\n",
              "       'Time from sample to castration resistance [months]', 'Somatic Status',\n",
              "       'SO comments', 'Survival Status', 'Tissue Site', 'TMB (nonsynonymous)',\n",
              "       'Tumor Purity'],\n",
              "      dtype='object')"
            ]
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df.info()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "edixuMbBUoWj",
        "outputId": "71e86fb8-ec41-495f-9516-5fad047557dc"
      },
      "execution_count": 10,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 424 entries, 0 to 423\n",
            "Data columns (total 40 columns):\n",
            " #   Column                                              Non-Null Count  Dtype  \n",
            "---  ------                                              --------------  -----  \n",
            " 0   Study ID                                            424 non-null    object \n",
            " 1   Patient ID                                          424 non-null    object \n",
            " 2   Sample ID                                           424 non-null    object \n",
            " 3   Age at Sample Collection                            424 non-null    object \n",
            " 4   Androgen Deprivation Therapy (ADT)                  421 non-null    object \n",
            " 5   Archer Panel                                        424 non-null    object \n",
            " 6   Biopsy Gleason Grade                                384 non-null    object \n",
            " 7   Cancer Type                                         424 non-null    object \n",
            " 8   Cancer Type Detailed                                424 non-null    object \n",
            " 9   Timing of metastases                                424 non-null    object \n",
            " 10  Disease Extent                                      424 non-null    object \n",
            " 11  Disease volume                                      424 non-null    object \n",
            " 12  Fraction Genome Altered                             424 non-null    float64\n",
            " 13  Gene Panel                                          424 non-null    object \n",
            " 14  Institute Source                                    424 non-null    object \n",
            " 15  Castration Resistance Event                         424 non-null    object \n",
            " 16  MSI Comment                                         221 non-null    object \n",
            " 17  MSI Score                                           424 non-null    float64\n",
            " 18  MSI Type                                            424 non-null    object \n",
            " 19  MSK Slide ID                                        334 non-null    float64\n",
            " 20  Mutation Count                                      399 non-null    float64\n",
            " 21  Oncotree Code                                       424 non-null    object \n",
            " 22  Overall Survival since Sample Collection (Months)   424 non-null    float64\n",
            " 23  12-245 Part A Consented                             424 non-null    object \n",
            " 24  12-245 Part C Consented                             424 non-null    object \n",
            " 25  MSK Pathology Slide Available                       424 non-null    object \n",
            " 26  Pediatric Case Indicator                            424 non-null    object \n",
            " 27  Prostate-specific antigen                           411 non-null    float64\n",
            " 28  Race Category                                       353 non-null    object \n",
            " 29  Sample Class                                        424 non-null    object \n",
            " 30  Number of Samples Per Patient                       424 non-null    int64  \n",
            " 31  Sample coverage                                     424 non-null    int64  \n",
            " 32  Sex                                                 424 non-null    object \n",
            " 33  Time from sample to castration resistance [months]  424 non-null    float64\n",
            " 34  Somatic Status                                      424 non-null    object \n",
            " 35  SO comments                                         132 non-null    object \n",
            " 36  Survival Status                                     424 non-null    object \n",
            " 37  Tissue Site                                         424 non-null    object \n",
            " 38  TMB (nonsynonymous)                                 424 non-null    float64\n",
            " 39  Tumor Purity                                        419 non-null    float64\n",
            "dtypes: float64(9), int64(2), object(29)\n",
            "memory usage: 132.6+ KB\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df.isna().sum()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "0FnA_DEoUwkP",
        "outputId": "43ca84c8-bafc-4acc-9143-2aedf9fc71f6"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Study ID                                                0\n",
              "Patient ID                                              0\n",
              "Sample ID                                               0\n",
              "Age at Sample Collection                                0\n",
              "Androgen Deprivation Therapy (ADT)                      3\n",
              "Archer Panel                                            0\n",
              "Biopsy Gleason Grade                                   40\n",
              "Cancer Type                                             0\n",
              "Cancer Type Detailed                                    0\n",
              "Timing of metastases                                    0\n",
              "Disease Extent                                          0\n",
              "Disease volume                                          0\n",
              "Fraction Genome Altered                                 0\n",
              "Gene Panel                                              0\n",
              "Institute Source                                        0\n",
              "Castration Resistance Event                             0\n",
              "MSI Comment                                           203\n",
              "MSI Score                                               0\n",
              "MSI Type                                                0\n",
              "MSK Slide ID                                           90\n",
              "Mutation Count                                         25\n",
              "Oncotree Code                                           0\n",
              "Overall Survival since Sample Collection (Months)       0\n",
              "12-245 Part A Consented                                 0\n",
              "12-245 Part C Consented                                 0\n",
              "MSK Pathology Slide Available                           0\n",
              "Pediatric Case Indicator                                0\n",
              "Prostate-specific antigen                              13\n",
              "Race Category                                          71\n",
              "Sample Class                                            0\n",
              "Number of Samples Per Patient                           0\n",
              "Sample coverage                                         0\n",
              "Sex                                                     0\n",
              "Time from sample to castration resistance [months]      0\n",
              "Somatic Status                                          0\n",
              "SO comments                                           292\n",
              "Survival Status                                         0\n",
              "Tissue Site                                             0\n",
              "TMB (nonsynonymous)                                     0\n",
              "Tumor Purity                                            5\n",
              "dtype: int64"
            ],
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>Study ID</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Patient ID</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Sample ID</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Age at Sample Collection</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Androgen Deprivation Therapy (ADT)</th>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Archer Panel</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Biopsy Gleason Grade</th>\n",
              "      <td>40</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Cancer Type</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Cancer Type Detailed</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Timing of metastases</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Disease Extent</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Disease volume</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Fraction Genome Altered</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Gene Panel</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Institute Source</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Castration Resistance Event</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MSI Comment</th>\n",
              "      <td>203</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MSI Score</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MSI Type</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MSK Slide ID</th>\n",
              "      <td>90</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Mutation Count</th>\n",
              "      <td>25</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Oncotree Code</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Overall Survival since Sample Collection (Months)</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12-245 Part A Consented</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12-245 Part C Consented</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MSK Pathology Slide Available</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Pediatric Case Indicator</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Prostate-specific antigen</th>\n",
              "      <td>13</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Race Category</th>\n",
              "      <td>71</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Sample Class</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Number of Samples Per Patient</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Sample coverage</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Sex</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Time from sample to castration resistance [months]</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Somatic Status</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>SO comments</th>\n",
              "      <td>292</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Survival Status</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Tissue Site</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TMB (nonsynonymous)</th>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Tumor Purity</th>\n",
              "      <td>5</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div><br><label><b>dtype:</b> int64</label>"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "for i in df.columns:\n",
        "  print(\"\\n==============================================\\n\" , df[i].value_counts() , \"\\n==============================================\\n\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5qcozGQgVuNF",
        "outputId": "e511edfe-5e20-4c4a-b93e-f20e42470edb"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "==============================================\n",
            " Study ID\n",
            "prad_mcspc_mskcc_2020    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Patient ID\n",
            "P-0033767    1\n",
            "P-0000373    1\n",
            "P-0000431    1\n",
            "P-0000442    1\n",
            "P-0000509    1\n",
            "            ..\n",
            "P-0001141    1\n",
            "P-0001128    1\n",
            "P-0001092    1\n",
            "P-0000888    1\n",
            "P-0000842    1\n",
            "Name: count, Length: 424, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Sample ID\n",
            "P-0033767-T01-IM6    1\n",
            "P-0000373-T01-IM3    1\n",
            "P-0000431-T01-IM3    1\n",
            "P-0000442-T01-IM3    1\n",
            "P-0000509-T01-IM3    1\n",
            "                    ..\n",
            "P-0001141-T01-IM3    1\n",
            "P-0001128-T01-IM3    1\n",
            "P-0001092-T01-IM3    1\n",
            "P-0000888-T01-IM3    1\n",
            "P-0000842-T01-IM3    1\n",
            "Name: count, Length: 424, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Age at Sample Collection\n",
            "66     24\n",
            "64     20\n",
            "71     20\n",
            "65     20\n",
            "56     18\n",
            "61     18\n",
            "67     17\n",
            "68     16\n",
            "73     15\n",
            "72     15\n",
            "63     15\n",
            "70     15\n",
            "74     14\n",
            "69     14\n",
            "60     13\n",
            "62     13\n",
            "58     12\n",
            "55     11\n",
            "54     11\n",
            "76     11\n",
            "75     10\n",
            "59      9\n",
            "<45     7\n",
            "77      7\n",
            "53      7\n",
            "57      6\n",
            "80      6\n",
            "52      6\n",
            ">85     6\n",
            "48      5\n",
            "81      5\n",
            "82      5\n",
            "51      5\n",
            "78      4\n",
            "49      4\n",
            "85      4\n",
            "84      4\n",
            "83      4\n",
            "50      3\n",
            "47      2\n",
            "45      1\n",
            "79      1\n",
            "46      1\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Androgen Deprivation Therapy (ADT)\n",
            "No     330\n",
            "Yes     91\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Archer Panel\n",
            "NO     413\n",
            "YES     11\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Biopsy Gleason Grade\n",
            "9-10    191\n",
            "8        88\n",
            "4+3      55\n",
            "3+4      30\n",
            "<7       19\n",
            "9-11      1\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Cancer Type\n",
            "Prostate Cancer    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Cancer Type Detailed\n",
            "Prostate Adenocarcinoma    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Timing of metastases\n",
            "De-novo metastatic       275\n",
            "Metastatic recurrence    149\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Disease Extent\n",
            "Metastatic hormone-sensitive    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Disease volume\n",
            "High-volume disease    213\n",
            "Low-volume disease     211\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Fraction Genome Altered\n",
            "0.0000    6\n",
            "0.0035    3\n",
            "0.1060    3\n",
            "0.0254    2\n",
            "0.0026    2\n",
            "         ..\n",
            "0.1303    1\n",
            "0.3625    1\n",
            "0.1053    1\n",
            "0.0325    1\n",
            "0.4323    1\n",
            "Name: count, Length: 393, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Gene Panel\n",
            "IMPACT468    233\n",
            "IMPACT410    154\n",
            "IMPACT341     37\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Institute Source\n",
            "MSKCC    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Castration Resistance Event\n",
            "Yes    240\n",
            "No     184\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " MSI Comment\n",
            "MICROSATELLITE STABLE (MSS). See MSI note below.                                                              172\n",
            "MICROSATELLITE STABLE (MSS). This result is for investigational use only. See MSI note below.                  38\n",
            "MICROSATELLITE INSTABILITY-HIGH (MSI-H). See MSI note below.                                                    8\n",
            "MICROSATELLITE INSTABILITY-INDETERMINATE. See MSI note below.                                                   2\n",
            "MICROSATELLITE INSTABILITY-INDETERMINATE. This result is for investigational use only. See MSI note below.      1\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " MSI Score\n",
            "0.00     85\n",
            "0.08     18\n",
            "0.07     11\n",
            "0.13     10\n",
            "0.14      9\n",
            "         ..\n",
            "22.38     1\n",
            "12.58     1\n",
            "0.48      1\n",
            "0.50      1\n",
            "1.08      1\n",
            "Name: count, Length: 144, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " MSI Type\n",
            "Stable           342\n",
            "Do not report     65\n",
            "Instable          12\n",
            "Indeterminate      5\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " MSK Slide ID\n",
            "1147146.0    1\n",
            "472020.0     1\n",
            "473451.0     1\n",
            "474027.0     1\n",
            "478159.0     1\n",
            "            ..\n",
            "482641.0     1\n",
            "745966.0     1\n",
            "745727.0     1\n",
            "745655.0     1\n",
            "745602.0     1\n",
            "Name: count, Length: 334, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Mutation Count\n",
            "2.0     94\n",
            "1.0     77\n",
            "3.0     75\n",
            "4.0     54\n",
            "5.0     35\n",
            "6.0     23\n",
            "7.0     14\n",
            "8.0      4\n",
            "10.0     3\n",
            "45.0     2\n",
            "12.0     2\n",
            "9.0      2\n",
            "42.0     1\n",
            "26.0     1\n",
            "24.0     1\n",
            "35.0     1\n",
            "44.0     1\n",
            "16.0     1\n",
            "41.0     1\n",
            "33.0     1\n",
            "65.0     1\n",
            "59.0     1\n",
            "76.0     1\n",
            "23.0     1\n",
            "39.0     1\n",
            "27.0     1\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Oncotree Code\n",
            "PRAD    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Overall Survival since Sample Collection (Months)\n",
            "31.30    3\n",
            "26.33    3\n",
            "17.03    3\n",
            "19.03    2\n",
            "21.00    2\n",
            "        ..\n",
            "18.13    1\n",
            "56.06    1\n",
            "33.48    1\n",
            "34.37    1\n",
            "28.06    1\n",
            "Name: count, Length: 390, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " 12-245 Part A Consented\n",
            "YES    422\n",
            "NO       2\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " 12-245 Part C Consented\n",
            "YES    355\n",
            "NO      69\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " MSK Pathology Slide Available\n",
            "YES    334\n",
            "NO      90\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Pediatric Case Indicator\n",
            "No    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Prostate-specific antigen\n",
            "6.0      25\n",
            "4.0      22\n",
            "5.0      22\n",
            "3.0      19\n",
            "7.0      16\n",
            "         ..\n",
            "95.0      1\n",
            "156.0     1\n",
            "397.0     1\n",
            "301.0     1\n",
            "85.0      1\n",
            "Name: count, Length: 167, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Race Category\n",
            "White                        312\n",
            "Black or African American     27\n",
            "Other                         14\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Sample Class\n",
            "Tumor    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Number of Samples Per Patient\n",
            "1    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Sample coverage\n",
            "816    4\n",
            "665    3\n",
            "740    3\n",
            "660    3\n",
            "808    3\n",
            "      ..\n",
            "742    1\n",
            "868    1\n",
            "375    1\n",
            "315    1\n",
            "264    1\n",
            "Name: count, Length: 324, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Sex\n",
            "Male    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Time from sample to castration resistance [months]\n",
            "0.00     6\n",
            "4.84     4\n",
            "12.68    3\n",
            "11.00    3\n",
            "7.10     3\n",
            "        ..\n",
            "14.94    1\n",
            "18.13    1\n",
            "40.67    1\n",
            "22.16    1\n",
            "54.68    1\n",
            "Name: count, Length: 376, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Somatic Status\n",
            "Matched    424\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " SO comments\n",
            "NANote: Low tumor content (approximately 20% or less). Negative mutation, copy number and structural variant results should be interpreted with caution.NA                                                                                                         30\n",
            "NANote: Low tumor content (approximately 20% or less).  Negative mutation, copy number and structural variant results, as well as low (stable) MSIsensor scores, should be interpreted with caution.NA                                                             10\n",
            "Assessment of the copy number profile is limited by increased background noise.False negatives cannot be ruled out. Confirmation of copy number changes by alternate methods is suggested if clinically indicated.                                                  7\n",
            "Note: Copy number profile is suggestive of broad copy number gain on Chromosome arm 8q.                                                                                                                                                                             6\n",
            "Note: Low tumor content (approximately 20% or less). Negative results should be interpreted with caution.                                                                                                                                                           6\n",
            "                                                                                                                                                                                                                                                                   ..\n",
            "NANote: Copy number profile is suggestive of broad copy number gains on Chromosome arms 3q22-28, 8q11-24.21 and 10q11-21. Note: Copy number profile is suggestive of broad copy number losses on Chromosome arms 10q, 17p and 17q11-21.31.NA                        1\n",
            "Note: Copy number profile is suggestive of broad copy number gains on Chromosome arms 8q and Chromosome 9.                                                                                                                                                          1\n",
            "NANote: Copy number profile is suggestive of broad copy number loss on Chromosome arms 1p36, 5q11-22, 6q15-22, 13q and 16q.NA                                                                                                                                       1\n",
            "NANote: Copy number profile is suggestive of multiple chromosome arm level losses, including RB1 on Chromosome arm 13q.NA                                                                                                                                           1\n",
            "NANote: Low tumor content (approximately 20% or less).  Negative structural variant results, as well as low (stable) MSIsensor scores, should be interpreted with caution.  Additional mutations and copy number alterations may not be detected due to low tum     1\n",
            "Name: count, Length: 71, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Survival Status\n",
            "Alive    322\n",
            "Dead     102\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Tissue Site\n",
            "Prostate             204\n",
            "Lymph Node            93\n",
            "Bone                  61\n",
            "Lung                  26\n",
            "Other soft tissue     25\n",
            "Liver                 15\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " TMB (nonsynonymous)\n",
            "1.729396     54\n",
            "2.594094     41\n",
            "1.957439     36\n",
            "0.978720     36\n",
            "3.458792     35\n",
            "0.864698     32\n",
            "0.000000     25\n",
            "4.323490     24\n",
            "2.936159     24\n",
            "3.914879     15\n",
            "4.893598     12\n",
            "3.327466     10\n",
            "6.052887     10\n",
            "5.872318     10\n",
            "1.109155      9\n",
            "5.188189      9\n",
            "2.218311      5\n",
            "6.851038      4\n",
            "6.654932      3\n",
            "4.436621      2\n",
            "7.782283      2\n",
            "6.917585      2\n",
            "8.646981      2\n",
            "7.829758      2\n",
            "13.309864     1\n",
            "49.911989     1\n",
            "7.764087      1\n",
            "11.744636     1\n",
            "34.255189     1\n",
            "23.489273     1\n",
            "25.446712     1\n",
            "36.317320     1\n",
            "9.787197      1\n",
            "43.063667     1\n",
            "13.835170     1\n",
            "35.452622     1\n",
            "28.535037     1\n",
            "56.205376     1\n",
            "51.017188     1\n",
            "65.717055     1\n",
            "19.888056     1\n",
            "38.911414     1\n",
            "34.587924     1\n",
            "23.346849     1\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n",
            "\n",
            "==============================================\n",
            " Tumor Purity\n",
            "30.0    81\n",
            "20.0    64\n",
            "40.0    63\n",
            "60.0    54\n",
            "50.0    50\n",
            "10.0    37\n",
            "80.0    28\n",
            "70.0    23\n",
            "15.0     8\n",
            "25.0     3\n",
            "90.0     3\n",
            "35.0     2\n",
            "75.0     1\n",
            "65.0     1\n",
            "95.0     1\n",
            "Name: count, dtype: int64 \n",
            "==============================================\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df.drop(columns=['SO comments' , 'MSI Comment' , 'MSK Slide ID' , 'Patient ID' , 'Study ID' , 'Sample ID' , 'Cancer Type' , 'Cancer Type Detailed'\n",
        "                 , 'Race Category'\n",
        "                 ,'Disease Extent'\n",
        "                 ,'Oncotree Code'\n",
        "                 ,'Pediatric Case Indicator',\n",
        "                 'Sample Class'\n",
        "                 ,'Number of Samples Per Patient'\n",
        "                 ,'Somatic Status'\n",
        "                 ,'Institute Source'\n",
        "                 ]\n",
        "        ,axis=1\n",
        "        ,inplace=True)"
      ],
      "metadata": {
        "id": "d3jGsNhWU1p1"
      },
      "execution_count": 15,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 499
        },
        "id": "W0GXcXiHW2eO",
        "outputId": "7e9743cd-3e4e-4c67-9174-84b23ab209c7"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "  Age at Sample Collection Androgen Deprivation Therapy (ADT) Archer Panel  \\\n",
              "0                       55                                 No           NO   \n",
              "1                       57                                Yes           NO   \n",
              "2                       67                                 No           NO   \n",
              "3                       64                                Yes           NO   \n",
              "4                       57                                Yes           NO   \n",
              "\n",
              "  Biopsy Gleason Grade Timing of metastases       Disease volume  \\\n",
              "0                 9-10   De-novo metastatic   Low-volume disease   \n",
              "1                    8   De-novo metastatic   Low-volume disease   \n",
              "2                    8   De-novo metastatic  High-volume disease   \n",
              "3                 9-10   De-novo metastatic   Low-volume disease   \n",
              "4                 9-10   De-novo metastatic   Low-volume disease   \n",
              "\n",
              "   Fraction Genome Altered Gene Panel Castration Resistance Event  MSI Score  \\\n",
              "0                   0.0023  IMPACT341                         Yes       0.09   \n",
              "1                   0.1985  IMPACT341                         Yes       0.34   \n",
              "2                   0.1419  IMPACT341                          No       0.18   \n",
              "3                   0.0161  IMPACT341                         Yes       0.17   \n",
              "4                   0.3134  IMPACT341                         Yes       0.36   \n",
              "\n",
              "   ... 12-245 Part C Consented  MSK Pathology Slide Available  \\\n",
              "0  ...                     YES                            YES   \n",
              "1  ...                     YES                            YES   \n",
              "2  ...                     YES                            YES   \n",
              "3  ...                     YES                            YES   \n",
              "4  ...                     YES                            YES   \n",
              "\n",
              "   Prostate-specific antigen Sample coverage   Sex  \\\n",
              "0                        6.0             708  Male   \n",
              "1                       11.0             530  Male   \n",
              "2                       90.0             922  Male   \n",
              "3                       21.0             651  Male   \n",
              "4                       32.0             515  Male   \n",
              "\n",
              "  Time from sample to castration resistance [months]  Survival Status  \\\n",
              "0                                              20.03             Dead   \n",
              "1                                              35.58            Alive   \n",
              "2                                              56.29            Alive   \n",
              "3                                              39.84            Alive   \n",
              "4                                              21.58             Dead   \n",
              "\n",
              "   Tissue Site TMB (nonsynonymous)  Tumor Purity  \n",
              "0   Lymph Node            1.109155          60.0  \n",
              "1   Lymph Node            1.109155          60.0  \n",
              "2     Prostate            1.109155          50.0  \n",
              "3     Prostate            2.218311          50.0  \n",
              "4     Prostate            3.327466          60.0  \n",
              "\n",
              "[5 rows x 24 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-dbe273f1-aa46-4b3e-b206-e8851b126218\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Age at Sample Collection</th>\n",
              "      <th>Androgen Deprivation Therapy (ADT)</th>\n",
              "      <th>Archer Panel</th>\n",
              "      <th>Biopsy Gleason Grade</th>\n",
              "      <th>Timing of metastases</th>\n",
              "      <th>Disease volume</th>\n",
              "      <th>Fraction Genome Altered</th>\n",
              "      <th>Gene Panel</th>\n",
              "      <th>Castration Resistance Event</th>\n",
              "      <th>MSI Score</th>\n",
              "      <th>...</th>\n",
              "      <th>12-245 Part C Consented</th>\n",
              "      <th>MSK Pathology Slide Available</th>\n",
              "      <th>Prostate-specific antigen</th>\n",
              "      <th>Sample coverage</th>\n",
              "      <th>Sex</th>\n",
              "      <th>Time from sample to castration resistance [months]</th>\n",
              "      <th>Survival Status</th>\n",
              "      <th>Tissue Site</th>\n",
              "      <th>TMB (nonsynonymous)</th>\n",
              "      <th>Tumor Purity</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>55</td>\n",
              "      <td>No</td>\n",
              "      <td>NO</td>\n",
              "      <td>9-10</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>Low-volume disease</td>\n",
              "      <td>0.0023</td>\n",
              "      <td>IMPACT341</td>\n",
              "      <td>Yes</td>\n",
              "      <td>0.09</td>\n",
              "      <td>...</td>\n",
              "      <td>YES</td>\n",
              "      <td>YES</td>\n",
              "      <td>6.0</td>\n",
              "      <td>708</td>\n",
              "      <td>Male</td>\n",
              "      <td>20.03</td>\n",
              "      <td>Dead</td>\n",
              "      <td>Lymph Node</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>60.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>57</td>\n",
              "      <td>Yes</td>\n",
              "      <td>NO</td>\n",
              "      <td>8</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>Low-volume disease</td>\n",
              "      <td>0.1985</td>\n",
              "      <td>IMPACT341</td>\n",
              "      <td>Yes</td>\n",
              "      <td>0.34</td>\n",
              "      <td>...</td>\n",
              "      <td>YES</td>\n",
              "      <td>YES</td>\n",
              "      <td>11.0</td>\n",
              "      <td>530</td>\n",
              "      <td>Male</td>\n",
              "      <td>35.58</td>\n",
              "      <td>Alive</td>\n",
              "      <td>Lymph Node</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>60.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>67</td>\n",
              "      <td>No</td>\n",
              "      <td>NO</td>\n",
              "      <td>8</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>High-volume disease</td>\n",
              "      <td>0.1419</td>\n",
              "      <td>IMPACT341</td>\n",
              "      <td>No</td>\n",
              "      <td>0.18</td>\n",
              "      <td>...</td>\n",
              "      <td>YES</td>\n",
              "      <td>YES</td>\n",
              "      <td>90.0</td>\n",
              "      <td>922</td>\n",
              "      <td>Male</td>\n",
              "      <td>56.29</td>\n",
              "      <td>Alive</td>\n",
              "      <td>Prostate</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>50.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>64</td>\n",
              "      <td>Yes</td>\n",
              "      <td>NO</td>\n",
              "      <td>9-10</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>Low-volume disease</td>\n",
              "      <td>0.0161</td>\n",
              "      <td>IMPACT341</td>\n",
              "      <td>Yes</td>\n",
              "      <td>0.17</td>\n",
              "      <td>...</td>\n",
              "      <td>YES</td>\n",
              "      <td>YES</td>\n",
              "      <td>21.0</td>\n",
              "      <td>651</td>\n",
              "      <td>Male</td>\n",
              "      <td>39.84</td>\n",
              "      <td>Alive</td>\n",
              "      <td>Prostate</td>\n",
              "      <td>2.218311</td>\n",
              "      <td>50.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>57</td>\n",
              "      <td>Yes</td>\n",
              "      <td>NO</td>\n",
              "      <td>9-10</td>\n",
              "      <td>De-novo metastatic</td>\n",
              "      <td>Low-volume disease</td>\n",
              "      <td>0.3134</td>\n",
              "      <td>IMPACT341</td>\n",
              "      <td>Yes</td>\n",
              "      <td>0.36</td>\n",
              "      <td>...</td>\n",
              "      <td>YES</td>\n",
              "      <td>YES</td>\n",
              "      <td>32.0</td>\n",
              "      <td>515</td>\n",
              "      <td>Male</td>\n",
              "      <td>21.58</td>\n",
              "      <td>Dead</td>\n",
              "      <td>Prostate</td>\n",
              "      <td>3.327466</td>\n",
              "      <td>60.0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 24 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-dbe273f1-aa46-4b3e-b206-e8851b126218')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-dbe273f1-aa46-4b3e-b206-e8851b126218 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-dbe273f1-aa46-4b3e-b206-e8851b126218');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-03d0ca48-3d1d-4e15-b001-3ea88bc8b168\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-03d0ca48-3d1d-4e15-b001-3ea88bc8b168')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-03d0ca48-3d1d-4e15-b001-3ea88bc8b168 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "df"
            }
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "x = df.drop('Disease volume' , axis =1)\n",
        "y = df['Disease volume']"
      ],
      "metadata": {
        "id": "r_a5qVanW3hB"
      },
      "execution_count": 17,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "from sklearn.model_selection import train_test_split\n",
        "from sklearn.preprocessing import LabelEncoder\n",
        "\n",
        "lb = LabelEncoder()\n",
        "y = lb.fit_transform(y)"
      ],
      "metadata": {
        "id": "MF708TGPXAkz"
      },
      "execution_count": 18,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "ob_cols = x.select_dtypes(include='object').columns"
      ],
      "metadata": {
        "id": "Xp3FLns9XMyi"
      },
      "execution_count": 19,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "for i in ob_cols:\n",
        "  most_freq = x[i].mode()[0]\n",
        "  x[i].fillna(most_freq , inplace=True)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WR_BcsycXRV4",
        "outputId": "bfcce17c-908c-442b-fdbd-9eaf4478fb64"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/tmp/ipython-input-1637567976.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.\n",
            "The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.\n",
            "\n",
            "For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.\n",
            "\n",
            "\n",
            "  x[i].fillna(most_freq , inplace=True)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "x = pd.get_dummies(x, columns=ob_cols, drop_first=True)\n",
        "display(x.head())"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 342
        },
        "id": "Kaz7VZhZXd3a",
        "outputId": "0df974e4-bd36-4d4c-ee2a-c126904baac6"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "   Fraction Genome Altered  MSI Score  Mutation Count  \\\n",
              "0                   0.0023       0.09             1.0   \n",
              "1                   0.1985       0.34             1.0   \n",
              "2                   0.1419       0.18             1.0   \n",
              "3                   0.0161       0.17             2.0   \n",
              "4                   0.3134       0.36             3.0   \n",
              "\n",
              "   Overall Survival since Sample Collection (Months)  \\\n",
              "0                                              38.75   \n",
              "1                                              59.48   \n",
              "2                                              56.29   \n",
              "3                                              69.50   \n",
              "4                                              46.90   \n",
              "\n",
              "   Prostate-specific antigen  Sample coverage  \\\n",
              "0                        6.0              708   \n",
              "1                       11.0              530   \n",
              "2                       90.0              922   \n",
              "3                       21.0              651   \n",
              "4                       32.0              515   \n",
              "\n",
              "   Time from sample to castration resistance [months]  TMB (nonsynonymous)  \\\n",
              "0                                              20.03              1.109155   \n",
              "1                                              35.58              1.109155   \n",
              "2                                              56.29              1.109155   \n",
              "3                                              39.84              2.218311   \n",
              "4                                              21.58              3.327466   \n",
              "\n",
              "   Tumor Purity  Age at Sample Collection_46  ...  MSI Type_Stable  \\\n",
              "0          60.0                        False  ...             True   \n",
              "1          60.0                        False  ...             True   \n",
              "2          50.0                        False  ...             True   \n",
              "3          50.0                        False  ...             True   \n",
              "4          60.0                        False  ...             True   \n",
              "\n",
              "   12-245 Part A Consented_YES  12-245 Part C Consented_YES  \\\n",
              "0                         True                         True   \n",
              "1                         True                         True   \n",
              "2                         True                         True   \n",
              "3                         True                         True   \n",
              "4                         True                         True   \n",
              "\n",
              "   MSK Pathology Slide Available_YES  Survival Status_Dead  Tissue Site_Liver  \\\n",
              "0                               True                  True              False   \n",
              "1                               True                 False              False   \n",
              "2                               True                 False              False   \n",
              "3                               True                 False              False   \n",
              "4                               True                  True              False   \n",
              "\n",
              "   Tissue Site_Lung  Tissue Site_Lymph Node  Tissue Site_Other soft tissue  \\\n",
              "0             False                    True                          False   \n",
              "1             False                    True                          False   \n",
              "2             False                   False                          False   \n",
              "3             False                   False                          False   \n",
              "4             False                   False                          False   \n",
              "\n",
              "   Tissue Site_Prostate  \n",
              "0                 False  \n",
              "1                 False  \n",
              "2                  True  \n",
              "3                  True  \n",
              "4                  True  \n",
              "\n",
              "[5 rows x 74 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-532c8a1c-7c92-4a50-b487-c885842385d2\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Fraction Genome Altered</th>\n",
              "      <th>MSI Score</th>\n",
              "      <th>Mutation Count</th>\n",
              "      <th>Overall Survival since Sample Collection (Months)</th>\n",
              "      <th>Prostate-specific antigen</th>\n",
              "      <th>Sample coverage</th>\n",
              "      <th>Time from sample to castration resistance [months]</th>\n",
              "      <th>TMB (nonsynonymous)</th>\n",
              "      <th>Tumor Purity</th>\n",
              "      <th>Age at Sample Collection_46</th>\n",
              "      <th>...</th>\n",
              "      <th>MSI Type_Stable</th>\n",
              "      <th>12-245 Part A Consented_YES</th>\n",
              "      <th>12-245 Part C Consented_YES</th>\n",
              "      <th>MSK Pathology Slide Available_YES</th>\n",
              "      <th>Survival Status_Dead</th>\n",
              "      <th>Tissue Site_Liver</th>\n",
              "      <th>Tissue Site_Lung</th>\n",
              "      <th>Tissue Site_Lymph Node</th>\n",
              "      <th>Tissue Site_Other soft tissue</th>\n",
              "      <th>Tissue Site_Prostate</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>0.0023</td>\n",
              "      <td>0.09</td>\n",
              "      <td>1.0</td>\n",
              "      <td>38.75</td>\n",
              "      <td>6.0</td>\n",
              "      <td>708</td>\n",
              "      <td>20.03</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>60.0</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>0.1985</td>\n",
              "      <td>0.34</td>\n",
              "      <td>1.0</td>\n",
              "      <td>59.48</td>\n",
              "      <td>11.0</td>\n",
              "      <td>530</td>\n",
              "      <td>35.58</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>60.0</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>0.1419</td>\n",
              "      <td>0.18</td>\n",
              "      <td>1.0</td>\n",
              "      <td>56.29</td>\n",
              "      <td>90.0</td>\n",
              "      <td>922</td>\n",
              "      <td>56.29</td>\n",
              "      <td>1.109155</td>\n",
              "      <td>50.0</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>0.0161</td>\n",
              "      <td>0.17</td>\n",
              "      <td>2.0</td>\n",
              "      <td>69.50</td>\n",
              "      <td>21.0</td>\n",
              "      <td>651</td>\n",
              "      <td>39.84</td>\n",
              "      <td>2.218311</td>\n",
              "      <td>50.0</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>0.3134</td>\n",
              "      <td>0.36</td>\n",
              "      <td>3.0</td>\n",
              "      <td>46.90</td>\n",
              "      <td>32.0</td>\n",
              "      <td>515</td>\n",
              "      <td>21.58</td>\n",
              "      <td>3.327466</td>\n",
              "      <td>60.0</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 74 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-532c8a1c-7c92-4a50-b487-c885842385d2')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-532c8a1c-7c92-4a50-b487-c885842385d2 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-532c8a1c-7c92-4a50-b487-c885842385d2');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-aaa7c76b-ef96-4e3e-8f5f-764464323e5d\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-aaa7c76b-ef96-4e3e-8f5f-764464323e5d')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-aaa7c76b-ef96-4e3e-8f5f-764464323e5d button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "x = x.fillna(x.mean())"
      ],
      "metadata": {
        "id": "6LFltZBqXx_X"
      },
      "execution_count": 22,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "x.isna().sum().sum()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wvdyoVYqX12e",
        "outputId": "621eb073-3752-4b57-865a-70dbe8b913b8"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "np.int64(0)"
            ]
          },
          "metadata": {},
          "execution_count": 24
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "print(x.shape , y.shape)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dKYQVS8OX4jE",
        "outputId": "8a537a59-ef45-4216-ea59-e27f3a4ba9e2"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "(424, 74) (424,)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from sklearn.preprocessing import StandardScaler\n",
        "sc = StandardScaler()\n",
        "x = sc.fit_transform(x)"
      ],
      "metadata": {
        "id": "FzImS2ZOYAaB"
      },
      "execution_count": 26,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "xtrain , xtest , ytrain , ytest = train_test_split(x , y , test_size=0.2 , random_state=42)"
      ],
      "metadata": {
        "id": "XlKx2LdLYRhC"
      },
      "execution_count": 27,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "from collections import Counter\n",
        "print(Counter(y))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9_TkrYnvYV3J",
        "outputId": "2e42f1cf-8c32-4cd3-f300-19f55d066f9e"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Counter({np.int64(0): 213, np.int64(1): 211})\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "xtrain.shape , xtest.shape"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QdvLtcJoYZ8n",
        "outputId": "07cd6cd1-3d55-47e8-9e77-042e1c4ca054"
      },
      "execution_count": 30,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "((339, 74), (85, 74))"
            ]
          },
          "metadata": {},
          "execution_count": 30
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from sklearn.svm import SVC\n",
        "from sklearn.model_selection import GridSearchCV\n",
        "from sklearn.metrics import classification_report\n",
        "\n",
        "param_grid = {'C': [0.001 ,0.1 ,0.5, 1, 10,15,50, 100], 'gamma': [1, 0.1, 0.01, 0.001], 'kernel': ['rbf', 'linear']}\n",
        "\n",
        "grid = GridSearchCV(SVC(), param_grid,cv = 5)\n",
        "grid.fit(xtrain, ytrain)\n",
        "\n",
        "print(\"Best parameters found: \", grid.best_params_)\n",
        "print(\"Best score found: \", grid.best_score_)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "D2_kjcHqYnVQ",
        "outputId": "31944e9e-9e48-489d-c9d5-9b338c9f00f1"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Best parameters found:  {'C': 1, 'gamma': 0.01, 'kernel': 'rbf'}\n",
            "Best score found:  0.7433274802458297\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "model = grid.best_estimator_\n",
        "mo = SVC(C=1, gamma=0.01, kernel='rbf')\n",
        "mo.fit(xtrain, ytrain)\n",
        "ypred = model.predict(xtest)\n",
        "ypred2 = mo.predict(xtest)\n",
        "\n",
        "print(classification_report(ytest, ypred,digits=4))\n",
        "print(classification_report(ytest, ypred2,digits=4))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "O6vY6C09Zb_R",
        "outputId": "eca6c24e-5a81-4c24-a98b-a57490924d82"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "              precision    recall  f1-score   support\n",
            "\n",
            "           0     0.7436    0.5918    0.6591        49\n",
            "           1     0.5652    0.7222    0.6341        36\n",
            "\n",
            "    accuracy                         0.6471        85\n",
            "   macro avg     0.6544    0.6570    0.6466        85\n",
            "weighted avg     0.6680    0.6471    0.6485        85\n",
            "\n",
            "              precision    recall  f1-score   support\n",
            "\n",
            "           0     0.7436    0.5918    0.6591        49\n",
            "           1     0.5652    0.7222    0.6341        36\n",
            "\n",
            "    accuracy                         0.6471        85\n",
            "   macro avg     0.6544    0.6570    0.6466        85\n",
            "weighted avg     0.6680    0.6471    0.6485        85\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from sklearn.tree import DecisionTreeClassifier\n",
        "\n",
        "param_grid_dt = {'max_depth': [None, 5, 10, 15, 20],\n",
        "                 'min_samples_split': [2, 5, 10],\n",
        "                 'min_samples_leaf': [1, 2, 4],\n",
        "                 'criterion': ['gini', 'entropy']}\n",
        "\n",
        "grid_dt = GridSearchCV(DecisionTreeClassifier(), param_grid_dt, cv=5, verbose=2)\n",
        "grid_dt.fit(xtrain, ytrain)\n",
        "\n",
        "print(\"Best parameters found for Decision Tree: \", grid_dt.best_params_)\n",
        "print(\"Best score found for Decision Tree: \", grid_dt.best_score_)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Kt1V9fzWaMBb",
        "outputId": "7cd616a9-326d-4113-8c2d-0378307e9ab3"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Fitting 5 folds for each of 90 candidates, totalling 450 fits\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=gini, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=None, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=5, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=10, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=15, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=1, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=2, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=2; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=5; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "[CV] END criterion=entropy, max_depth=20, min_samples_leaf=4, min_samples_split=10; total time=   0.0s\n",
            "Best parameters found for Decision Tree:  {'criterion': 'entropy', 'max_depth': 15, 'min_samples_leaf': 1, 'min_samples_split': 2}\n",
            "Best score found for Decision Tree:  0.7285776997366111\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "model_dt = grid_dt.best_estimator_\n",
        "ypred_dt = model_dt.predict(xtest)\n",
        "\n",
        "print(classification_report(ytest, ypred_dt,digits=4))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PoRRXydjdrc_",
        "outputId": "5121288e-a849-4a01-eaa4-6eeb8c56f1be"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "              precision    recall  f1-score   support\n",
            "\n",
            "           0     0.7045    0.6327    0.6667        49\n",
            "           1     0.5610    0.6389    0.5974        36\n",
            "\n",
            "    accuracy                         0.6353        85\n",
            "   macro avg     0.6328    0.6358    0.6320        85\n",
            "weighted avg     0.6437    0.6353    0.6373        85\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from sklearn.ensemble import RandomForestClassifier\n",
        "\n",
        "param_grid_rf = {'n_estimators': [100, 200, 500],\n",
        "                 'max_depth': [None, 5, 10, 15],\n",
        "                 'min_samples_split': [2, 5, 10],\n",
        "                 'min_samples_leaf': [1, 2, 4],\n",
        "                 'criterion': ['gini', 'entropy']}\n",
        "\n",
        "grid_rf = GridSearchCV(RandomForestClassifier(random_state=42), param_grid_rf, cv=5, verbose=2, n_jobs=-1)\n",
        "grid_rf.fit(xtrain, ytrain)\n",
        "\n",
        "print(\"Best parameters found for Random Forest: \", grid_rf.best_params_)\n",
        "print(\"Best score found for Random Forest: \", grid_rf.best_score_)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qqZCA5Jad1ne",
        "outputId": "808a4fc2-4542-45c6-ad1a-7a22cd2fb83a"
      },
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Fitting 5 folds for each of 216 candidates, totalling 1080 fits\n",
            "Best parameters found for Random Forest:  {'criterion': 'gini', 'max_depth': None, 'min_samples_leaf': 2, 'min_samples_split': 5, 'n_estimators': 100}\n",
            "Best score found for Random Forest:  0.7493415276558385\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "model_rf = grid_rf.best_estimator_\n",
        "ypred_rf = model_rf.predict(xtest)\n",
        "\n",
        "print(classification_report(ytest, ypred_rf,digits=4))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "x7YCpo0EeHrp",
        "outputId": "f5da3625-8319-470b-a52c-3443e5377974"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "              precision    recall  f1-score   support\n",
            "\n",
            "           0     0.8649    0.6531    0.7442        49\n",
            "           1     0.6458    0.8611    0.7381        36\n",
            "\n",
            "    accuracy                         0.7412        85\n",
            "   macro avg     0.7553    0.7571    0.7411        85\n",
            "weighted avg     0.7721    0.7412    0.7416        85\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c87fdb29",
        "outputId": "1f66af7c-5abb-4667-9eb5-5904a49d71b8"
      },
      "source": [
        "!pip install xgboost"
      ],
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: xgboost in /usr/local/lib/python3.12/dist-packages (3.0.5)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from xgboost) (2.0.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12 in /usr/local/lib/python3.12/dist-packages (from xgboost) (2.27.3)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.12/dist-packages (from xgboost) (1.16.2)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3f8690d4",
        "outputId": "6e5ec3fb-362a-4a33-9dd5-cd2f3ad8e515"
      },
      "source": [
        "from xgboost import XGBClassifier\n",
        "from sklearn.model_selection import GridSearchCV\n",
        "from sklearn.metrics import classification_report\n",
        "\n",
        "param_grid_xgb = {\n",
        "    'n_estimators': [100, 200, 500],\n",
        "    'max_depth': [3, 5, 7],\n",
        "    'learning_rate': [0.01, 0.1, 0.2],\n",
        "    'subsample': [0.6, 0.8, 1.0],\n",
        "    'colsample_bytree': [0.6, 0.8, 1.0]\n",
        "}\n",
        "\n",
        "grid_xgb = GridSearchCV(XGBClassifier(random_state=42), param_grid_xgb, cv=5, verbose=2, n_jobs=-1)\n",
        "grid_xgb.fit(xtrain, ytrain)\n",
        "\n",
        "print(\"Best parameters found for XGBoost: \", grid_xgb.best_params_)\n",
        "print(\"Best score found for XGBoost: \", grid_xgb.best_score_)"
      ],
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Fitting 5 folds for each of 243 candidates, totalling 1215 fits\n",
            "Best parameters found for XGBoost:  {'colsample_bytree': 0.8, 'learning_rate': 0.01, 'max_depth': 3, 'n_estimators': 200, 'subsample': 1.0}\n",
            "Best score found for XGBoost:  0.749297629499561\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "model_xgb = grid_xgb.best_estimator_\n",
        "ypred_xgb = model_xgb.predict(xtest)\n",
        "\n",
        "print(classification_report(ytest, ypred_xgb,digits=4))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9hWIp-sZgn3g",
        "outputId": "04cbfe13-99d0-430c-ff1f-9a19caeb513f"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "              precision    recall  f1-score   support\n",
            "\n",
            "           0     0.8611    0.6327    0.7294        49\n",
            "           1     0.6327    0.8611    0.7294        36\n",
            "\n",
            "    accuracy                         0.7294        85\n",
            "   macro avg     0.7469    0.7469    0.7294        85\n",
            "weighted avg     0.7644    0.7294    0.7294        85\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "G-ZG_xyKhS-a"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}